### PR TITLE
Migrate to GitHub Issues new forms

### LINKED ISSUE #1342: Enhancement: Migrate to GitHub Issues forms
### Implementation PR

#1343

### Reference Issues

_No response_

### Summary

GitHub Issues is rolling out the new forms for PR.

They look like this issue looks. It allows for easier formatting and validation using YAML.

### Basic Example

image
This is an example of opening a new issue.

### Drawbacks and Impact

None.

### Unresolved questions

_No response_

### Additional Context

_No response_

### Contribution Guidelines

- [X] I agree to follow Starlite's Contribution Guidelines

### Code of Conduct

- [X] I agree to follow Starlite's Code of Conduct

### PR DESCRIPTION
**Pull request by JacobCoffee**
### Pull Request Checklist

[//]: # "Please review the [Starlite contribution guidelines](https://github.com/starlite-api/starlite/blob/main/CONTRIBUTING.rst) for this repository."

- [x] Have you followed the guidelines in the [Starlite Contribution Guidelines](https://github.com/starlite-api/starlite/blob/main/CONTRIBUTING.rst)?
- [ ] Have you got 100% test coverage on new code?
- [ ] Have you updated the prose documentation?
- [ ] Have you updated the reference documentation?

### Description

[//]: # "Please describe your pull request for new release changelog purposes"
- Migrate issue templates to new YAML-based forms
- Update pull request template (to this template you are reading right now)

### Close Issue(s)

[//]: # "Please add in issue numbers this pull request will close, if applicable"
[//]: # "Examples: Fixes #4321 or Closes #1234"
Closes #1342

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/DOCS.yaml [hunk 1]
```diff
@@ -0,0 +1,53 @@
+name: "Feature Request"
+description: Create an issue for documentation changes
+title: "Docs: <title>"
+labels: ["docs", "triage required"]
```
Comment (cofin) [inline .github/ISSUE_TEMPLATE/DOCS.yaml:4]:
```suggestion
labels: ["docs"]
```
Do we need `triage required` on all the template types?

File: .github/ISSUE_TEMPLATE/REQUEST.yaml [hunk 2]
```diff
@@ -0,0 +1,77 @@
+name: "Feature Request"
+description: Create an issue for a new feature request
+title: "Enhancement: <title>"
+labels: ["enhancement", "triage required"]
```
Comment (cofin) [inline .github/ISSUE_TEMPLATE/REQUEST.yaml:4]:
```suggestion
labels: ["enhancement"]
```

Same as above.

**Review by cofin — APPROVED** [review]
LGTM.  Added a suggestion for input tags.  I'm ok with us keeping it there if needed though.

**Comment by JacobCoffee** [comment]
@paging the rest of the team since it affects UX in how users/we create issues.

**Comment by cofin** [comment]
Should `pre-commit` be updated here?  The failure doesn't look to be related to the changes.

**Comment by JacobCoffee** [comment]
@cofin i was thinking it would better be updated in a PR that references the issue/PR that broke it? but we can do it here if wanted?

File: .github/ISSUE_TEMPLATE/BUG.yaml [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+    attributes:
+      label: "Platform"
+      description: What platform are you encountering the issue on?
+      options:
+        - label: "Linux"
+        - label: "Mac"
+        - label: "Windows"
+        - label: "Other (Please specify in the description above)"
+    validations:
+      required: true
```
Comment (Goldziher) [inline .github/ISSUE_TEMPLATE/BUG.yaml:87]: i dont think this should be required

File: .github/ISSUE_TEMPLATE/BUG.yaml [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+  - type: checkboxes
```
Comment (Goldziher) [inline .github/ISSUE_TEMPLATE/BUG.yaml:88]: this is for a bug issue, why are these required?

File: .github/ISSUE_TEMPLATE/BUG.yaml [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+      description: Please copy and paste any relevant log output. This will be automatically formatted into code, so no need for backticks.
+      render: bash
+    validations:
+      required: false
+  - type: checkboxes
+    id: version
+    attributes:
+      label: "Starlite Version"
+      description: What version of Starlite are you using when encountering this issue?
+      options:
```
Comment (Goldziher) [inline .github/ISSUE_TEMPLATE/BUG.yaml:70]: i wouldnt use options here - it should be a simple input

File: .github/ISSUE_TEMPLATE/BUG.yaml [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+body:
+  - type: textarea
+    id: description
+    attributes:
+      label: "Description"
+      description: Please enter an description of the bug you are encountering
+      placeholder:
+    validations:
+      required: true
+  - type: input
```
Comment (Goldziher) [inline .github/ISSUE_TEMPLATE/BUG.yaml:14]: This is seriously over complicated i think. Just a simple test aread for a description/code etc. and an optional field to upload stuff is enough. 

IF the user does not need to go into all of these, fine, but I am really averse to repositories that ask me to fill complex forms to file an issue (as a user myself).

File: .github/ISSUE_TEMPLATE/DOCS.yaml [hunk 7]
```diff
@@ -0,0 +1,53 @@
+name: "Documentation Update"
+description: Create an issue for documentation changes
+title: "Docs: <title>"
+labels: ["docs"]
+body:
+  - type: textarea
+    id: implementation_pr
+    attributes:
+      label: "Implementation PR"
```
Comment (Goldziher) [inline .github/ISSUE_TEMPLATE/DOCS.yaml:9]: do we need this? github link via mentions etc.

File: .github/ISSUE_TEMPLATE/DOCS.yaml [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+      description: Link to the PR that implements the documentation change, if applicable
+      placeholder: "#PullRequestID"
+    validations:
+      required: false
+  - type: textarea
+    id: reference_issues
```
Comment (Goldziher) [inline .github/ISSUE_TEMPLATE/DOCS.yaml:15]: same

File: .github/ISSUE_TEMPLATE/DOCS.yaml [hunk 9]
```diff
@@ -0,0 +1,53 @@
...
+      required: true
+  - type: textarea
+    id: basic_example
+    attributes:
+      label: "Basic Example"
+      description: Indicate here some basic examples of your feature.
+      placeholder: Provide some basic example of your feature request
+    validations:
+      required: false
+  - type: checkboxes
```
Comment (Goldziher) [inline .github/ISSUE_TEMPLATE/DOCS.yaml:38]: so, since this is an issue template i dont see why we need this here.

File: .github/ISSUE_TEMPLATE/REQUEST.yaml [hunk 10]
```diff
@@ -0,0 +1,77 @@
+name: "Feature Request"
+description: Create an issue for a new feature request
+title: "Enhancement: <title>"
+labels: ["enhancement"]
+body:
+  - type: textarea
+    id: implementation_pr
```
Comment (Goldziher) [inline .github/ISSUE_TEMPLATE/REQUEST.yaml:7]: same points as prev for this and the next entry

File: .github/ISSUE_TEMPLATE/REQUEST.yaml [hunk 11]
```diff
@@ -0,0 +1,77 @@
...
+      required: false
+  - type: textarea
+    id: additional_context
+    attributes:
+      label: "Additional Context"
+      description: Add any other context or screenshots about the feature request here.
+      placeholder: Identify any other context about the feature request here.
+    validations:
+      required: false
+  - type: checkboxes
```
Comment (Goldziher) [inline .github/ISSUE_TEMPLATE/REQUEST.yaml:62]: same

**Comment by JacobCoffee** [comment]
I have updated to remove a lot of the required things so users can just go in and fill out basic things if desired.
Here is the outlook:

Menu
image

Feature/Enhancements
image

Docs
image

Bug Reports
image
image

thoughts?

**Comment by JacobCoffee** [comment]
Once submitted this form compiles into markdown and is not always a bulky ugly form.
But the form prompts users to help be more descriptive in their requests/issues.

image

**Review by Goldziher — APPROVED** [review]
